### PR TITLE
[SPARK-40769][CORE][SQL] Migrate type check failures of aggregate expressions onto error classes

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr aims to replace `TypeCheckFailure` by `DataTypeMismatch` in type checks in the aggregate expressions:

- Count
- CollectSet
- CountMinSketchAgg
- HistogramNumeric

### Why are the changes needed?
Migration onto error classes unifies Spark SQL error messages.

### Does this PR introduce _any_ user-facing change?
Yes. The PR changes user-facing error messages.

### How was this patch tested?
Pass GitHub Actions

### CODE REVIEW

**Comment by LuciferYang** [comment]
still in wip

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Count.scala [hunk 1]
```diff
@@ -56,9 +59,17 @@ case class Count(children: Seq[Expression]) extends DeclarativeAggregate {
...
-        s"If you have to call the function $prettyName without arguments, set the legacy " +
-        s"configuration `${SQLConf.ALLOW_PARAMETERLESS_COUNT.key}` as true")
+      DataTypeMismatch(
+        errorSubClass = "WRONG_NUM_ARGS_WITH_SUGGESTION",
+        messageParameters = Map(
+          "functionName" -> toSQLId(prettyName),
+          "expectedNum" -> " >= 1",
+          "actualNum" -> "0",
+          "legacyNum" -> "0",
+          "legacyConfKey" -> SQLConf.ALLOW_PARAMETERLESS_COUNT.key,
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Count.scala:69]: Please, use `toSQLConf()` to quote the config.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Count.scala [hunk 2]
```diff
@@ -56,9 +59,17 @@ case class Count(children: Seq[Expression]) extends DeclarativeAggregate {
...
+          "legacyConfValue" -> true.toString
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Count.scala:70]: `true` should be quoted by `""`, see `error-classes.json`. For instance,
https://github.com/apache/spark/blob/5741d38ee272418a919fe7d102514c221a6e741a/core/src/main/resources/error/error-classes.json#L10

I would add new helper `toSQLConfVal()` like:
```scala
  def toSQLConfVal(conf: String): String = {
    quoteByDefault(conf)
  }
```

**Review by MaxGekk — CHANGES_REQUESTED** [review]

**Review by MaxGekk — APPROVED** [review]
Waiting for CI.

**Comment by LuciferYang** [comment]
GA passed

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @LuciferYang.

**Pull request closed without merging**